pallets/oracle-feed/src/tests.rs

```diff
@@ -0,0 +1,55 @@
+use cfg_traits::ValueProvider;
+use frame_support::assert_ok;
+
+use crate::mock::*;
+
+const FEEDER: AccountId = 1;
+const KEY: OracleKey = 23;
+const VALUE1: OracleValue = 42;
+const VALUE2: OracleValue = 43;
+const TIMESTAMP1: Timestamp = 1234;
+const TIMESTAMP2: Timestamp = 1235;
+
+#[test]
+fn feed() {
+	new_test_ext().execute_with(|| {
+		MockTime::mock_now(|| TIMESTAMP1);
+		MockPayFee::mock_pay(|account| {
+			assert_eq!(*account, FEEDER);
+			Ok(())
+		});
+
+		assert_ok!(OracleFeed::feed(RuntimeOrigin::signed(FEEDER), KEY, VALUE1));
+		assert_ok!(
+			OracleFeed::get(&Some(FEEDER), &KEY),
+			Some((VALUE1, TIMESTAMP1))
+		);
+
+		MockTime::mock_now(|| TIMESTAMP2);
+		MockPayFee::mock_pay(|_| unreachable!("Feeding the same key again do not require fees"));
+
+		assert_ok!(OracleFeed::feed(RuntimeOrigin::signed(FEEDER), KEY, VALUE2));
+		assert_ok!(
+			OracleFeed::get(&Some(FEEDER), &KEY),
+			Some((VALUE2, TIMESTAMP2))
+		);
+	});
+}
+
+#[test]
+fn feed_root() {
+	new_test_ext().execute_with(|| {
+		MockTime::mock_now(|| TIMESTAMP1);
+		MockPayFee::mock_pay(|_| unreachable!("Feeding from root does not require fees"));
+
+		assert_ok!(OracleFeed::feed(RuntimeOrigin::root(), KEY, VALUE1));
+		assert_ok!(OracleFeed::get(&None, &KEY), Some((VALUE1, TIMESTAMP1)));
+	});
+}
+
```

Comment:
When changing the key of `FedValues` from `T::AccountId` to `T::RuntimeOrigin`, we need to add another test which ensures the `None` origin cannot feed (because it cannot be charged).

Reply:
Finally, I've left it under an `EnsureOrigin` trait, which should determine what to do in each case. So, the test is not needed now because it is handled in the runtime. But thanks for this very accurate comment! 🙌🏻